File: app/src/main/java/za/co/gundula/app/arereyeng/sync/AreYengAuthenticatorService.java
package za.co.gundula.app.arereyeng.sync;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.Nullable;

/**
 * Created by kgundula on 2016/12/07.
 */

public class AreYengAuthenticatorService extends Service {

    private AreYengAuthenticator mAuthenticator;

    @Override
    public void onCreate() {
        // Create a new authenticator object
        mAuthenticator = new AreYengAuthenticator(this);
    }


    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return mAuthenticator.getIBinder();
    }
}

File: app/src/main/java/za/co/gundula/app/arereyeng/sync/AreYengAuthenticatorService.kt
package za.co.gundula.app.arereyeng.sync

import android.app.Service
import android.content.Intent
import android.os.IBinder

/**
 * Created by kgundula on 2016/12/07.
 */
class AreYengAuthenticatorService : Service() {
    private var mAuthenticator: AreYengAuthenticator? = null
    override fun onCreate() {
        // Create a new authenticator object
        mAuthenticator = AreYengAuthenticator(this)
    }

    override fun onBind(intent: Intent): IBinder? {
        return mAuthenticator!!.iBinder
    }
}